publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java

```diff
@@ -396,18 +396,28 @@ public List<Map<String, Object>> processingValidEvent(String eventResponseMessag
         Map<String, Object> eventResponse = null;
 
         if (eventResponseMessage == null) {
-            eventResponse.put("Parameter 'eventResponseMessage' must not be null", HttpStatus.BAD_REQUEST);
+            eventResponse.put("Missing response from 'generate' service", HttpStatus.BAD_REQUEST);
             responseEvent.add(eventResponse);
             return responseEvent;
         }
         JsonElement eventElement = JsonParser.parseString(eventResponseMessage);
         JsonArray eventArray = eventElement.getAsJsonArray();
+        if (eventArray == null) {
+            String errorMessage = "Invalid, response json event is not in the form of JSON array";
+            log.error(errorMessage);
+            eventResponse.put(errorMessage, HttpStatus.BAD_REQUEST);
+        }
         for (int i = 0; i < eventArray.size(); i++) {
             eventResponse = new HashMap<>();
             JsonElement jsonResponseEvent = eventArray.get(i);
             if (isValid(jsonResponseEvent)) {
                 synchronized (this) {
                     JsonObject validResponse = jsonResponseEvent.getAsJsonObject();
+                    if (validResponse == null) {
+                        String errorMessage = "Invalid, response json event is not in the form of JSON object";
```

Comment:
Misleading error message, use something like "Invalid, array item expected to be in the form of JSON object".